### PR TITLE
Typescript strict mode

### PR DESCRIPTION
**Pull request by ivan-dalmet**
Remove all errors with `strict: true` in `tsconfig.json` file

### CODE REVIEW

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]
First little review.

File: src/components/DataList/index.tsx [hunk 1]
```diff
@@ -8,33 +8,43 @@ import {
...
 
 import { useDarkMode } from '@/hooks/useDarkMode';
 
-export const DataListContext = React.createContext(null);
-export const DataListHeaderContext = React.createContext(null);
+type DataListColumns = Record<string, DataListCellProps>;
+type DataListContextValue = {
+  setColumns: React.Dispatch<React.SetStateAction<DataListColumns>>;
+  columns: DataListColumns;
+  isHover;
```
Comment (yoannfleurydev) [inline src/components/DataList/index.tsx:22]: No type for `isHover` ?

**Review by HugoPerard — APPROVED** [review]

File: src/components/DataList/index.tsx [hunk 2]
```diff
@@ -8,33 +8,43 @@ import {
...
+};
+type DataListHeaderContextValue = boolean;
+
+export const DataListContext = React.createContext<DataListContextValue>(
+  {} as TODO
```
Comment (DecampsRenan) [inline src/components/DataList/index.tsx:27]:
```suggestion
  {} as Partial<DataListContextValue>
```

**Review by DecampsRenan — APPROVED** [review]

File: src/components/DataList/index.tsx (hunk 2 above)
Comment (ivan-dalmet) [reply src/components/DataList/index.tsx:27]: This will not work ;)

**Review by yoannfleurydev — APPROVED** [review]

**Pull request merged**